# AP Physics C: Resistance Investigation

**Research Question:** How does current vary with varying resistance and constant voltage?

## Data Collection

A variety of resistors of resistance from 100 Ohms to 500 were used in the investigation. The results were as follows:

In [2]:
import pandas as pd
import numpy as np

In [3]:
data = np.array([
    [100, 0.0623],
    [200, 0.0319],
    [300, 0.022],
    [400, 0.0169],
    [500, 0.0137]
])

In [4]:
df = pd.DataFrame(data, columns = ["Resistance", "Current"])

In [5]:
df

Resistance  Current
0       100.0   0.0623
1       200.0   0.0319
2       300.0   0.0220
3       400.0   0.0169
4       500.0   0.0137

In [6]:
R = df["Resistance"]
I = df["Current"]

In [7]:
def reciprocal_model(x, a):
    return a / x

## Data Analysis

A line of best fit was generated by `curve_fit` from SciPy, and plotted in addition to the actual data points collected. A reciprocal model $f(x) = \frac{a}{x}$ was used.

In [8]:
import plotly.graph_objects as go
from scipy.optimize import curve_fit
from plotly.offline import plot, iplot, init_notebook_mode
init_notebook_mode(connected=True)

In [9]:
param, param_cov = curve_fit(reciprocal_model, R, I)

In [10]:
samples = 160
R_generated = np.linspace(100, 500, num=samples)

In [11]:
I_best_fit = reciprocal_model(R_generated, param[0])

In [12]:
fig1 = go.Figure()
fig1.add_trace(go.Scatter(x=R_generated, y=I_best_fit, mode="lines", name="Best fit"))
fig1.add_trace(go.Scatter(x=R, y=I, mode="markers", name="Data values"))
fig1.update_layout(title="Current from Resistance at V = 6")
fig1.update_layout(
    xaxis_title=r"$R (\Omega)$",
    yaxis_title=r"$I (A)$"
)
fig1.show()

As the model fit an inverse relation better, a plot of $\frac{1}{R}$ by $I$ was also generated:

In [13]:
fig1 = go.Figure()
fig1.add_trace(go.Scatter(x=1 / R_generated, y=I_best_fit, mode="lines", name="Best fit"))
fig1.add_trace(go.Scatter(x=1 / R, y=I, mode="markers", name="Data values"))
fig1.update_layout(title="Current from inverse resistance at V = 6")
fig1.update_layout(
    xaxis_title=r"$R (\Omega)$",
    yaxis_title=r"$I (A)$"
)
fig1.show()

In [79]:
param[0]

6.3232871511074

Within the model, the parameter used was $a = 6.32$, as compared to an actual nominal value of 6 volts. From the model, it can be deduced that:

$$
m = \left(\frac{I_2 - I_1}{\frac{1}{R_2} - \frac{1}{R_1}}
\right) = 6.32 \, A\Omega = 6.32 \, V
$$

$$
R \propto \frac{1}{I}
$$

$$
R = \frac{a}{I} \Rightarrow R = \frac{V}{I}
$$

The percent error from an actual value of V = 6V was calculated as follows:

In [85]:
def percent_err(expected, actual):
    err = abs(expected - actual) / actual * 100
    return "{:2.1f}%".format(err)

In [86]:
percent_err(param[0], 6)

'5.4%'

Thus, it can be seen that results were collected to an accuracy of 5.4%.